tests/components/fressnapf_tracker/conftest.py

```diff
@@ -0,0 +1,167 @@
+
+from fressnapftracker import (
+    Device,
+    PhoneVerificationResponse,
+    Position,
+    SmsCodeResponse,
+    Tracker,
+    TrackerFeatures,
+    TrackerSettings,
+    UserToken,
+)
+import pytest
+
+from homeassistant.components.fressnapf_tracker.const import (
+    CONF_ACCESS_TOKEN,
+    CONF_PHONE_NUMBER,
+    CONF_USER_ID,
+    DOMAIN,
+)
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+MOCK_PHONE_NUMBER = "+491234567890"
+MOCK_USER_ID = 12345
+MOCK_ACCESS_TOKEN = "mock_access_token"
+MOCK_SERIAL_NUMBER = "ABC123456"
+MOCK_DEVICE_TOKEN = "mock_device_token"
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock]:
+    """Override async_setup_entry."""
+    with patch(
+        "homeassistant.components.fressnapf_tracker.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        yield mock_setup_entry
+
+
+@pytest.fixture
+def mock_tracker() -> Tracker:
+    """Create a mock Tracker object."""
+    return Tracker(
+        name="Fluffy",
+        battery=85,
+        charging=False,
+        position=Position(
+            lat=52.520008,
+            lng=13.404954,
+            accuracy=10,
+            timestamp="2024-01-15T12:00:00Z",
+        ),
+        tracker_settings=TrackerSettings(
+            generation="GPS Tracker 2.0",
+            features=TrackerFeatures(live_tracking=True),
+        ),
+        led_brightness=None,
+        deep_sleep=None,
+        led_activatable=None,
+    )
+
+
+@pytest.fixture
+def mock_tracker_no_position() -> Tracker:
+    """Create a mock Tracker object without position."""
+    return Tracker(
+        name="Fluffy",
+        battery=85,
+        charging=False,
+        position=None,
+        tracker_settings=TrackerSettings(
+            generation="GPS Tracker 2.0",
+            features=TrackerFeatures(live_tracking=True),
+        ),
+        led_brightness=None,
+        deep_sleep=None,
+        led_activatable=None,
+    )
+
+
+@pytest.fixture
+def mock_device() -> Device:
+    """Create a mock Device object."""
+    return Device(
+        serialnumber=MOCK_SERIAL_NUMBER,
+        token=MOCK_DEVICE_TOKEN,
+    )
+
+
+@pytest.fixture
+def mock_auth_client(mock_device: Device) -> Generator[MagicMock]:
+    """Mock the AuthClient."""
+    with patch(
+        "homeassistant.components.fressnapf_tracker.config_flow.AuthClient"
+    ) as mock_auth_client:
+        client = mock_auth_client.return_value
+        client.request_sms_code = AsyncMock(
+            return_value=SmsCodeResponse(id=MOCK_USER_ID)
+        )
+        client.verify_phone_number = AsyncMock(
+            return_value=PhoneVerificationResponse(
+                user_token=UserToken(
+                    access_token=MOCK_ACCESS_TOKEN,
+                    refresh_token=None,
+                )
+            )
+        )
+        client.get_devices = AsyncMock(return_value=[mock_device])
+        yield client
```

Comment:
You can merge this one with `mock_init_auth_client`

```suggestion
@pytest.fixture
def mock_auth_client(mock_device: Device) -> Generator[MagicMock]:
    """Mock the AuthClient."""
    with (patch(
        "homeassistant.components.fressnapf_tracker.config_flow.AuthClient", autospec=True
    ) as mock_auth_client,
    patch(
        "homeassistant.components.fressnapf_tracker.AuthClient", new=mock_auth_client
    )):
        client = mock_auth_client.return_value
        client.request_sms_code = AsyncMock(
            return_value=SmsCodeResponse(id=MOCK_USER_ID)
        )
        client.verify_phone_number = AsyncMock(
            return_value=PhoneVerificationResponse(
                user_token=UserToken(
                    access_token=MOCK_ACCESS_TOKEN,
                    refresh_token=None,
                )
            )
        )
        client.get_devices = AsyncMock(return_value=[mock_device])
        yield client
```